### PR TITLE
Product listing for each category + Product mongoose schema

### LINKED ISSUE #16: Designing  pages of the list of products for each type
Someone should design the list of items for each times of product we have 

image

For example, In amazon, this is the list of electronics, there are some apple accessories, some tv box, etc. 

We all have to do a list like this for each type of product we have. 

For example for electronics, there will be some computers, games, phones etc

We will have to do this for each type of products (electronics, books etc)

### LINKED ISSUE #32: Create Product database model
Create a Mongoose schema for products according to the API design document. Part of user story 2.

### LINKED ISSUE #34: User Story#1: Login/register customer
As a customer, I want to be able to create a new account and/or log in to an existing account because I want to be able to shop for products.

Tasks: 
- Design and implement login page form.
- Design and implement register page form and save account information.
- Design and implement a list of products page.
- Design and implement a homepage of the website.

Tasks References : #29 #16 #17 #10 #39 #22 #38 

Acceptance tests:

Positive case
1. The customer clicks ‘Login’ button.
2. Login form appears.
3. The customer enters correct information and clicks ‘Login’ button.
4.  The customer clicks on their desired category.
5. A list of available products related to the clicked category and their description is shown.

Negative case
1. The customer clicks ‘Login’ button.
2. Login form appears.
3. The customer enters incorrect information and clicks ‘Login’ button.
4. An error message is displayed.

### PR DESCRIPTION
**Pull request by zihao429**
This PR adds the following
- A list of products for each categories that the store has #16. Each product contains its own unique information such as its product name, price, description, and stock.
- Added more images in the front page slideshow #10 
- A product/search API endpoint (might not be the correct way of doing it but can be refactored in another sprint)
- Fixed the logo bug in the navbar  #10 
- Adding the product mongoose schema #32 (Not sure if it was done properly)

Note : npm install axios must be downloaded in the client for everything to run correctly.

### CODE REVIEW

File: src/server/models/products.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var mongoose = require('mongoose');
```
Comment (MichaelDjabauri) [inline src/server/models/products.js:1]: Not sure about this either but as long as it works, maybe ask @Iwuh?

Comment (imran1289-ah) [reply src/server/models/products.js:1]: Yea i was not sure about this also

Comment (Iwuh) [reply src/server/models/products.js:1]: This looks fine, you have to require mongoose in each model file to get access to mongoose.Schema (mongoose is the package that lets you access the MongoDB database)

**Comment by Iwuh** [comment]
I just want to mention that while we can store static images on the client for now, I think we'll eventually want to store each product image in the database as we talked about during the meetings.

**Comment by imran1289-ah** [comment]
I resolved the conflicts. I still need someone to check if the conflicts were resolved correctly and that i didnt remove anything that was important

**Review by Iwuh — APPROVED** [review]
The backend changes look good to me so I'll give my approval, could @MichaelDjab or someone else check the frontend changes in commit a8de1e4 and make sure everything looks good?

**Review by MichaelDjabauri — APPROVED** [review]
This looks good. Eventually, the product data will be coming from the database.

**Pull request merged**